zst/cutter.go

```diff
@@ -42,50 +41,34 @@ type CutAssembler struct {
 	root    *column.IntReader
 	builder zcode.Builder
 	leaf    string
-	cuts    map[int]cut
+	cuts    map[int]typedReader
 }
 
-type cut struct {
+type typedReader struct {
 	typ    zed.Type
 	reader column.Reader
-	depth  int
 }
 
-func NewCutAssembler(zctx *zed.Context, fields []string, object *Object) (*CutAssembler, error) {
-	a := object.assembly
-	root, err := column.NewIntReader(a.root, object.seeker)
-	if err != nil {
-		return nil, err
-	}
-	cuts := make(map[int]cut)
-	for k, typ := range a.types {
-		recType := zed.TypeRecordOf(typ)
-		if typ == nil {
-			// We could be smarter here and just ignore values that
-			// aren't records but all this will be subsumed when we
-			// work on predicate pushdown, so no sense spending time
-			// on this right now.
-			return nil, fmt.Errorf("ZST cut requires all top-level records to be records: encountered type %s", zson.FormatType(typ))
-		}
-		reader, err := column.NewRecordReader(recType, a.maps[k], object.seeker)
-		if err != nil {
-			return nil, err
+func NewCutAssembler(zctx *zed.Context, path field.Path, object *Object) (*CutAssembler, error) {
+	root := column.NewIntReader(object.root, object.seeker)
```

Comment:
Nit: This looks like it's used only in the CutAssembler composite literal, so move it down there.